Doc/library/importlib.rst

```diff
@@ -597,20 +604,20 @@ ABC hierarchy::
         .. deprecated:: 3.3
            This method is deprecated in favour of :meth:`path_stats`.  You don't
            have to implement it, but it is still available for compatibility
-           purposes. Raise :exc:`OSError` if the path cannot be handled.
+           purposes. If the *path* cannot be handled, this raises an
+           :exc:`OSError` or a :exc:`ValueError` depending on the reason.
 
         .. versionchanged:: 3.4
-           Raise :exc:`OSError` instead of :exc:`NotImplementedError`.
+           Raise :exc:`OSError` by default instead of :exc:`NotImplementedError`.
 
     .. method:: set_data(path, data)
 
         Optional abstract method which writes the specified bytes to a file
         path. Any intermediate directories which do not exist are to be created
         automatically.
 
-        When writing to the path fails because the path is read-only
-        (:const:`errno.EACCES`/:exc:`PermissionError`), do not propagate the
-        exception.
+        When writing to the path fails by raising an :class:`OSError`
+        or a :class:`ValueError`, the exception is not propagated.
```

Comment:
This change is not correct, only a lack of write permissions should be silently ignored, everything else should be propagated.

```suggestion
        When writing to the path fails because the path is read-only
        (:const:`errno.EACCES`/:exc:`PermissionError`), do not propagate the
        exception. Other exceptions should still be propagated.
```

Reply:
Since the tests are failing, should we also protect against `FileExistsError` when writing the data?